packages/agents/src/tests/client.test.ts

```diff
@@ -0,0 +1,540 @@
+        host: "localhost:1999"
+      });
+
+      expect(client).toBeDefined();
+      expect(client.agent).toBe("test-agent"); // Should convert to kebab-case
+      expect(client.name).toBe("default");
+    });
+
+    it("should be able to connect to an agent with options", () => {
+      const client = new AgentClient({
+        agent: "TestAgent",
+        name: "my-instance",
+        host: "custom.host.com"
+      });
+
+      expect(client).toBeDefined();
+      expect(client.agent).toBe("test-agent");
+      expect(client.name).toBe("my-instance");
+    });
+
+    it("should handle different agent name formats", () => {
+      const testCases = [
+        { input: "TestAgent", expected: "test-agent" },
+        { input: "test-agent", expected: "test-agent" },
+        { input: "TEST_AGENT", expected: "test-agent" },
+        { input: "testAgent", expected: "test-agent" }
+      ];
+
+      testCases.forEach(({ input, expected }) => {
+        const client = new AgentClient({
+          agent: input,
+          host: "localhost:1999"
+        });
+        expect(client.agent).toBe(expected);
+      });
+    });
+  });
+
+  describe("State Management", () => {
+    it("should sync state from client to server", () => {
+      const onStateUpdate = vi.fn();
+      const client = new AgentClient({
+        agent: "TestAgent",
+        host: "localhost:1999",
+        onStateUpdate
+      });
+
+      const sendSpy = vi.spyOn(client, "send");
+      const newState = { count: 5, name: "test" };
+
+      client.setState(newState);
+
+      expect(sendSpy).toHaveBeenCalledWith(
+        JSON.stringify({ state: newState, type: "cf_agent_state" })
+      );
+      expect(onStateUpdate).toHaveBeenCalledWith(newState, "client");
+    });
+
+    it("should receive state updates from server", () => {
+      const onStateUpdate = vi.fn();
+      const client = new AgentClient({
+        agent: "TestAgent",
+        host: "localhost:1999",
+        onStateUpdate
+      });
+
+      // Simulate server state update
+      const serverState = { count: 10, serverData: true };
+      const event = new MessageEvent("message", {
+        data: JSON.stringify({
+          type: "cf_agent_state",
+          state: serverState
+        })
+      });
+
+      client.dispatchEvent(event);
+
+      expect(onStateUpdate).toHaveBeenCalledWith(serverState, "server");
+    });
+
+    it("should handle complex state objects", () => {
+      const onStateUpdate = vi.fn();
+      const client = new AgentClient<{
+        user: { id: number; name: string };
+        settings: { theme: string };
+      }>({
+        agent: "TestAgent",
+        host: "localhost:1999",
+        onStateUpdate
+      });
+
+      const complexState = {
+        user: { id: 1, name: "Alice" },
+        settings: { theme: "dark" }
+      };
+
+      client.setState(complexState);
+
+      expect(onStateUpdate).toHaveBeenCalledWith(complexState, "client");
+    });
+  });
```

Comment:
we should have a test here with 2 clients connected to the same agent, and sending a state update on one should trigger onStateUpdate on the other